### PR TITLE
Feature/brocade coa

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description

Implement wired RADIUS CoA
Also implement SNMP LLDP check for phones based on Cisco Catalyst
# Impacts

Wired deauthentication of endpoints on Brocade switches
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Brocade wired switch RADIUS CoA (requires 08.30.d firmware at least)

### CODE REVIEW

**Comment by fdurand** [comment]
Does CoA works with mac-auth and 802.1x ?

**Comment by dwlfrth** [comment]
Tested on both macauth and 802.1x

**Comment by dwlfrth** [comment]
Review me

File: lib/pf/Switch/Brocade.pm [hunk 1]
```diff
@@ -198,6 +200,158 @@ sub isVoIPEnabled {
...
+=cut
+
+sub wiredeauthTechniques {
+    my ($self, $method, $connection_type) = @_;
+    my $logger = $self->logger;
+    if ($connection_type == $WIRED_802_1X) {
+        my $default = $SNMP::RADIUS;
+        my %tech = (
+            $SNMP::SNMP => 'dot1xPortReauthenticate',
+            $SNMP::RADIUS => 'deauthenticateMacRadius',
```
Comment (louismunro) [inline lib/pf/Switch/Brocade.pm:216]: Someday there will no longer be something like $SNMP::RADIUS.
I am adding refactoring that to my wishlist for v6.

File: lib/pf/Switch/Brocade.pm [hunk 2]
```diff
@@ -198,6 +200,158 @@ sub isVoIPEnabled {
...
+    # if can't SNMP read abort
+    return if ( !$self->connectRead() );
+
+    my $ifDescr = $self->getIfDesc($ifIndex);
+    return if (!defined($ifDescr) || $ifDescr eq '');
+
+    my $oid_lldpLocPortDesc = '1.0.8802.1.1.2.1.3.7.1.4'; # from LLDP-MIB
+
+    $logger->trace("SNMP get_table for lldpLocPortDesc: $oid_lldpLocPortDesc");
+    my $result = $self->{_sessionRead}->get_table( -baseoid => $oid_lldpLocPortDesc);
```
Comment (louismunro) [inline lib/pf/Switch/Brocade.pm:336]: Can the result be cached like we do for NasPortIdToIfIndex on Cisco Switches?

**Comment by louismunro** [comment]
Apart from the caching support, I see no problem merging.

**Comment by dwlfrth** [comment]
Cache added

**Comment by louismunro** [comment]
All your bases are belong to us then.
Merging.

**Pull request merged**